test/trafficsplit/trafficsplit_test.go

```diff
@@ -0,0 +1,262 @@
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+const ZeroRps = "0.0rps"
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+type statTsRow struct {
+	name       string
+	apex       string
+	leaf       string
+	weight     string
+	success    string
+	rps        string
+	latencyP50 string
+	latencyP95 string
+	latencyP99 string
+}
+
+func parseStatTsRow(out string, expectedRowCount, expectedColumnCount int) (map[string]*statTsRow, error) {
+	rows, err := testutil.CheckRowCount(out, expectedRowCount)
+	if err != nil {
+		return nil, err
+	}
+
+	statRows := make(map[string]*statTsRow)
+
+	for _, row := range rows {
+		fields := strings.Fields(row)
+
+		if len(fields) != expectedColumnCount {
+			return nil, fmt.Errorf(
+				"Expected [%d] columns in stat output, got [%d]; full output:\n%s",
+				expectedColumnCount, len(fields), row)
+		}
+
+		row := &statTsRow{
+			name:       fields[0],
+			apex:       fields[1],
+			leaf:       fields[2],
+			weight:     fields[3],
+			success:    fields[4],
+			rps:        fields[5],
+			latencyP50: fields[6],
+			latencyP95: fields[7],
+			latencyP99: fields[8],
+		}
+
+		statRows[row.leaf] = row
+
+	}
+	return statRows, nil
+}
+
+func statTrafficSplit(from string, ns string) (map[string]*statTsRow, error) {
+	cmd := []string{"stat", "ts", "--from", from, "--namespace", ns}
+	stdOut, _, err := TestHelper.LinkerdRun(cmd...)
+	if err != nil {
+		return nil, err
+	}
+	return parseStatTsRow(stdOut, 2, 9)
+}
+
+func validateTrafficSplit(actual *statTsRow, expected *statTsRow) error {
+
+	if actual.name != expected.name && expected.name != "" {
+		return fmt.Errorf("expected name '%s' for leaf %s, got '%s'", expected.name, expected.leaf, actual.name)
+	}
+
+	if actual.apex != expected.apex && expected.apex != "" {
+		return fmt.Errorf("expected apex '%s' for leaf %s, got '%s'", expected.apex, expected.leaf, actual.apex)
+	}
+
+	if actual.weight != expected.weight && expected.weight != "" {
+		return fmt.Errorf("expected weight '%s' for leaf %s, got '%s'", expected.weight, expected.leaf, actual.weight)
+	}
+
+	if actual.success != expected.success && expected.success != "" {
+		return fmt.Errorf("expected success '%s' for leaf %s, got '%s'", expected.success, expected.leaf, actual.success)
+	}
+
+	if expected.rps != "-" {
+		if expected.rps == ZeroRps && actual.rps != ZeroRps {
+			return fmt.Errorf("expected rps 0.0rps for leaf %s, got '%s'", expected.leaf, actual.rps)
+		}
+
+		if expected.rps != ZeroRps && actual.rps == ZeroRps {
+			return fmt.Errorf("expected rps '%s' for leaf %s, got '0.0rps'", expected.rps, expected.leaf)
+		}
+	} else {
+		if actual.rps != expected.rps && expected.rps != "" {
+			return fmt.Errorf("expected rps '%s' for leaf %s, got '%s'", expected.rps, expected.leaf, actual.rps)
+		}
+	}
+
+	if actual.latencyP50 != expected.latencyP50 && expected.latencyP50 != "" {
```

Comment:
we never compare latencies so I think we can remove this

Reply:
They are there so we can ensure that when there is no traffic observed at all, we have a placeholder such as: 

```
expectedFailingSvcOutput := &statTsRow{
  name:       "backend-traffic-split",
  apex:       "backend-svc",
  leaf:       "backend-svc",
  weight:     "0",
  success:    "-",
  rps:        "-",
  latencyP50: "-",
  latencyP95: "-",
  latencyP99: "-",
} 
```
We can get rid of that if you think its not worth comparing it.